src/canisters/frontend/ic-asset/src/asset_config.rs

```diff
@@ -100,82 +119,48 @@ impl AssetSourceDirectoryConfiguration {
-struct InterimAssetConfigRule {
-    r#match: String,
-    cache: Option<CacheConfig>,
-    #[serde(default, deserialize_with = "deser_headers")]
-    headers: Maybe<HeadersConfig>,
-    ignore: Option<bool>,
-}
-
-impl<T> Default for Maybe<T> {
-    fn default() -> Self {
-        Self::Absent
-    }
-}
 
-fn deser_headers<'de, D>(deserializer: D) -> Result<Maybe<HeadersConfig>, D::Error>
-where
-    D: serde::Deserializer<'de>,
-{
-    match serde_json::value::Value::deserialize(deserializer)? {
-        Value::Object(v) => Ok(Maybe::Value(
-            v.into_iter()
-                .map(|(k, v)| (k, v.to_string().trim_matches('"').to_string()))
-                .collect::<HashMap<String, String>>(),
-        )),
-        Value::Null => Ok(Maybe::Null),
-        _ => Err(serde::de::Error::custom(
-            "wrong data format for field `headers` (only map or null are allowed)",
-        )),
-    }
-}
+    /// Returns a collection of unused configuration objects from all `.ic-assets.json` files
+    pub fn get_unused_configs(&self) -> HashMap<PathBuf, Vec<AssetConfigRule>> {
+        let mut hm = HashMap::new();
+        // aggregate
+        for node in self.config_map.values() {
+            let config_node = &node.lock().unwrap();
 
-impl AssetConfigRule {
-    fn from_interim(
-        InterimAssetConfigRule {
-            r#match,
-            cache,
-            headers,
-            ignore,
-        }: InterimAssetConfigRule,
-        config_file_parent_dir: &Path,
-    ) -> anyhow::Result<Self> {
-        let glob = Glob::new(
-            config_file_parent_dir
-                .join(&r#match)
-                .to_str()
-                .with_context(|| {
-                    format!(
-                        "cannot combine {} and {} into a string (to be later used as a glob pattern)",
-                        config_file_parent_dir.display(),
-                        r#match
-                    )
-                })?,
-        )
-        .with_context(|| format!("{} is not a valid glob pattern", r#match))?.compile_matcher();
+            let origin = config_node.origin.clone();
 
-        Ok(Self {
-            r#match: glob,
-            cache,
-            headers,
-            ignore,
-        })
+            for rule in config_node.rules.clone() {
+                if !rule.used {
+                    hm.entry(origin.clone())
+                        .and_modify(|v: &mut Vec<AssetConfigRule>| v.push(rule.clone()))
+                        .or_insert_with(|| vec![rule.clone()]);
+                }
+            }
+        }
+        // dedup and remove full path from "match" field
+        for (path, rules) in hm.iter_mut() {
+            rules.sort_by_key(|v| v.r#match.glob().to_string());
+            rules.dedup();
+            for mut rule in rules {
+                let prefix_path = format!("{}/", path.display());
+                let original_glob = rule.r#match.glob().to_string().replace(&prefix_path, "");
```

Comment:
It would be more exact to remove this only from the beginning of the string, as `PathBuf.strip_prefix()` does

Reply:
sorry, misunderstood

Reply:
i went with `replace` to skip dealing with `Option` returned by `strip_prefix`

Reply:
Would you like to reconsider?
> remove this only from the beginning of the string

From `String.replace` docs:
> Replaces all matches of a pattern with another string.

Reply:
reconsidered :)